### PR TITLE
Support downsample in dataset

### PR DESCRIPTION
**Pull request by nikmo33**
As discussed briefly in https://github.com/nerfstudio-project/nerfstudio/pull/957 - it seems like have downsample support in the base InputDataset could be beneficial. While colmap outputs images at multiple-resolutions this may not be the case for many other applications where there are images only stored at a single resolution. This allows us to load in data in the raw format and downsample once loaded to disk

### CODE REVIEW

File: nerfstudio/data/datamanagers/base_datamanager.py [hunk 1]
```diff
@@ -266,6 +266,10 @@ class VanillaDataManagerConfig(InstantiateConfig):
     camera_optimizer: CameraOptimizerConfig = CameraOptimizerConfig()
     """Specifies the camera pose optimizer used during training. Helpful if poses are noisy, such as for data from
     Record3D."""
+    dataset_camera_scale_factor: float = 1.0
```
Comment (tancik) [inline nerfstudio/data/datamanagers/base_datamanager.py:269]: Maybe `camera_res_scale_factor`. I worry this may be ambiguous with extrinsic scale.

**Review by tancik — COMMENTED** [review]
This looks fine to me. I think we should still suggest that people pre-scale their images. We actually used to have this feature and found that people would upload 10mp images and everything would grind to a halt :)

**Comment by nikmo33** [comment]
> This looks fine to me. I think we should still suggest that people pre-scale their images. We actually used to have this feature and found that people would upload 10mp images and everything would grind to a halt :)

Fair enough. Agree - it is probably a judgement call on which of the above methods are appropriate. Let me know if you agree that this feature in additional also makes sense

File: nerfstudio/data/datamanagers/base_datamanager.py (hunk 1 above)
Comment (nikmo33) [reply nerfstudio/data/datamanagers/base_datamanager.py]: done

**Review by tancik — APPROVED** [review]
LGTM, I think having the option is generally a good thing.

**Pull request merged**